File: leetcode/weekly-contest/20240317/100248-字符串及其反转中是否存在同一子字符串.cpp
#include <string>
#include <unordered_set>

using std::string;
using std::unordered_set;

class Solution
{
public:
  bool isSubstringPresent(const string &s)
  {
    unordered_set<string> set;
    int n = s.length();
    for (int i = 0; i < n - 1; ++i) {
      string rs = s.substr(i, 2);
      set.emplace(rs);
      reverse(rs.begin(), rs.end());
      if (set.count(rs) > 0) {
        return true;
      }
    }
    return false;
  }
};

File: problem-sets/leetcode-weekly-contest/20240317/100248-字符串及其反转中是否存在同一子字符串.py
class Solution:
    def isSubstringPresent(self, s: str) -> bool:
        n = len(s)
        visit = set()
        for i in range(n - 1):
            sub = s[i:i+2]
            visit.add(sub[:])
            sub = sub[::-1]
            if sub in visit:
                return True
            
        return False